contracts/TokenDistributor.sol

```diff
@@ -0,0 +1,212 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.19;
+
+import {StakingRewardsV2} from "./StakingRewardsV2.sol";
+import {RewardEscrowV2} from "./RewardEscrowV2.sol";
+import {IKwenta} from "./interfaces/IKwenta.sol";
+import {ITokenDistributor} from "./interfaces/ITokenDistributor.sol";
+
+contract TokenDistributor is ITokenDistributor {
+    /// @notice represents the status of if a person already
+    /// claimed their epoch
+    mapping(address => mapping(uint => bool)) internal claimedEpochs;
+
+    /// @notice kwenta interface
+    IKwenta private kwenta;
+
+    /// @notice rewards staking contract
+    StakingRewardsV2 private stakingRewardsV2;
+
+    /// @notice escrow contract which holds (and may stake) reward tokens
+    RewardEscrowV2 private rewardEscrowV2;
+
+    /// @notice last recorded balance of KWENTA in contract
+    uint internal lastTokenBalance;
+
+    /// @notice last checkpoint time
+    uint internal lastCheckpoint;
+
+    /// @notice starting week of deployment
+    uint internal startTime;
+
+    /// @inheritdoc ITokenDistributor
+    uint[1000000000000000] public tokensPerEpoch;
```

Comment:
**Thought/Issue:** Is it correct that currently `tokensPerEpoch` follows the following format:
```
startOfEpochTimestamp => numberOfTokensForGivenEpoch
```

I find it slightly unnerving that we use the timestamp as the key, rather than  what you would expect `epochNumber` to be.
It seems prone to errors of slightly incorrect timestamps than using an actual epoch number - i.e. epoch 1, epoch 2, epoch 3 etc.
Instead it is more like:
```
epochNumber 0 = startingTimestamp
epochNumber 1 = startingTimestamp + 604800
epochNumber 2 = startingTimestamp + 1209600
epochNumber 4 = startingTimestamp + 1814400
```
And so on.

I would prefer it if the epochNumbers being used were just "1, 2, 3, 4 etc.".